src/main/java/com/google/cloud/healthcare/etl/runner/dicomtofhir/DicomToFhirStreamingRunner.java

```diff
@@ -0,0 +1,241 @@
+            context.output(webpath);
+            DicomToFhirStreamingRunner.LOG.info("Extracted webpath");
+        }
+    }
+
+    /**
+     * A DoFn that will take the response of the Study Metadata Read call from the DICOM API and reformat it into an
+     * input to be consumed by the mapping library.
+     */
+    public static class ReformatMappingFnInputString extends DoFn<String, String> {
+        @ProcessElement
+        public void processElement(DoFn<String, String>.ProcessContext context) {
+            String dicomMetadataResponse = context.element();
+            Gson gson = new Gson();
+            JsonArray jsonArray = gson.fromJson(dicomMetadataResponse, JsonArray.class);
+            JsonObject jsonObject = new JsonObject();
+            jsonObject.add("study", jsonArray);
+            context.output(gson.toJson(jsonObject));
+            DicomToFhirStreamingRunner.LOG.info("Sending instance to be mapped");
+        }
+    }
+
+    /**
+     * A DoFn that will take the response of the mapping library and wrap it into a FHIR bundle to be written to the
+     * FHIR store.
+     * TODO(b/174594428): Add a unique ID for each ImagingStudy FHIR resource to be uploaded to prevent creation of multiple FHIR resources for each ImagingStudy.
+     */
+    public static class CreateFhirResourceBundle extends DoFn<String, String> {
+        private static final String RequestMethod = "PUT";
+        private static final String InnerResourceType = "ImagingStudy";
+
+        @ProcessElement
+        public void processElement(DoFn<String, String>.ProcessContext context) {
+            String mappingOutputString = context.element();
+
+            Gson gson = new Gson();
+            JsonObject mappingOutput = gson.fromJson(mappingOutputString, JsonObject.class);
+            JsonObject requestObj = new JsonObject();
+
+            requestObj.addProperty("method", RequestMethod);
+            requestObj.addProperty("url", InnerResourceType);
+            JsonObject entryObj = new JsonObject();
+            entryObj.add("resource", mappingOutput);
+            entryObj.add("request", requestObj);
+            JsonArray entries = new JsonArray();
+            entries.add(entryObj);
+            JsonObject jsonObject = new JsonObject();
+            jsonObject.addProperty("resourceType", "Bundle");
+            jsonObject.addProperty("type", "batch");
+            jsonObject.add("entry", entries);
+            context.output(gson.toJson(jsonObject));
+            DicomToFhirStreamingRunner.LOG.info("Uploading to FHIR store");
+        }
+    }
+
+    public static void main(String[] args) {
+        Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+        Pipeline p = Pipeline.create(options);
+
+        MappingFn<HclsApiDicomMappableMessage> mappingFn = MappingFn.of(options.getMappingPath());
+
+        DicomIO.ReadStudyMetadata.Result dicomResult = p
```

Comment:
for each stage, for example here: can you add a comment like: "Reading pubsub and Retrieving DICOMStudyMetadata for instance"?